src/tlv/TlvString.ts

```diff
@@ -42,9 +42,10 @@ export class StringSchema<T extends TlvType.ByteString | TlvType.Utf8String> ext
         return reader.readPrimitive(typeLength) as TlvToPrimitive[T];
     }
 
-    override validate({ length }: TlvToPrimitive[T]): void {
-        if (length > this.maxLength) throw new Error(`String is too long: ${length}, max ${this.maxLength}.`);
-        if (length < this.minLength) throw new Error(`String is too short: ${length}, min ${this.minLength}.`);
+    override validate(value: TlvToPrimitive[T]): void {
+        if (typeof value !== "string" && !(value instanceof Uint8Array) && !Buffer.isBuffer(value)) throw new Error(`Expected string, got ${typeof value}.`);
```

Comment:
This is incorrect since the value type is linked to this.type:

if (this.type === TlvType.Utf8String && typeof value !== "string") throw ...
if (this.type === TlvType.ByteString && !(value instanceof ByteArray)) throw ...

Reply:
Thank you , makes sense